service/matching/taskQueueManager.go

```diff
@@ -605,3 +613,12 @@ func (c *taskQueueManagerImpl) QueueID() *taskQueueID {
 func (c *taskQueueManagerImpl) TaskQueueKind() enumspb.TaskQueueKind {
 	return c.taskQueueKind
 }
+
+func (c *taskQueueManagerImpl) checkRecentPollers() bool {
```

Comment:
instead of define a new method, I would just do it in AddTask method directly as:
```
if !c.HasPollerAfter(time.Now().Add(-noPollerThreshold)) {
	c.metricScope.IncCounter(metrics.NoRecentPollerTasksPerTaskQueueCounter)
}

```